stellarobservatory/quorum_intersection.py

```diff
@@ -0,0 +1,62 @@
+"""Torstens's quorum intersection checker (a Lachowski variant)"""
+import logging
+from typing import Callable, Set, Type, Tuple
+
+from stellarobservatory.quorums import greatest_quorum
+
+
+def quorum_intersection(fbas: Tuple[Callable[[Set[Type], Type], bool], Set[Type]]):
+    """Takes an FBAS with set of nodes V and returns True iff F has quorum intersection.
+    It prints two disjoint quorums otherwise."""
+    is_slice_contained, all_nodes = fbas
+    len_all_nodes = len(all_nodes)
+    for quorum in traverse_min_quorums(is_slice_contained, set(), all_nodes, len_all_nodes):
+        greatest_q = greatest_quorum(is_slice_contained, all_nodes.difference(quorum), set())
+        if greatest_q != set():
+            logging.info("Found two disjoint quorums: %s, %s", quorum, greatest_q)
+            return False
```

Comment:
I don't think we should print anything here but instead do `return False, quorum, greatest_q`

Reply:
OK 👍

Reply:
Bonus points: in the tests you could also check whether the returned sets are quorums and don't have intersections :)

Reply:
b2dcea3